### PR TITLE
fix: typos in documentation files

### PR DESCRIPTION
**Pull request by leopardracer**
This pull request contains changes to improve clarity, correctness and structure.

**Description correction:**
Corrected "Karnot" to "Kakarot".
Corrected "wriiten" to "written".

Please review the changes and let me know if any additional changes are needed.

### CODE REVIEW

**Comment by leopardracer** [comment]
@swetshaw Hi! Apologies for the inconvenience. I just wanted to check if there’s anything I can improve in my pull request. Please feel free to ask any questions or let me know if any changes are needed to meet the project's merge requirements. I’m happy to make adjustments!

File: docs/starknet/architecture/understanding-kakarot.md [hunk 1]
```diff
@@ -188,6 +188,6 @@ sustainable.
 ## Starknet MultiVM appchains, powered by Kakarot
 
 KKRT Labs will be working closely with trusted partners such as Starkware,
-Karnot and other appchain providers to deliver EVM-compatible Starknet
+Kakarot and other appchain providers to deliver EVM-compatible Starknet
```
Comment (Eikix) [inline docs/starknet/architecture/understanding-kakarot.md]: this isn't a typo, Karnot is one of our devops partners

File: docs/starknet/ecosystem/sdks/foundry.md [hunk 2]
```diff
@@ -66,4 +66,4 @@ $ forge verify-contract <your_contract_address> <path-to-contract>:<contract-nam
 --etherscan-api-key "kakarot_starknet_sepolia" \
 --num-of-optimizations 200 \
 --compiler-version "v0.8.26+commit.8a97fa7a"
-```
\ No newline at end of file
+```
```
Comment (Eikix) [inline docs/starknet/ecosystem/sdks/foundry.md:69]: what is the change made here?

File: docs/starknet/architecture/understanding-kakarot.md (hunk 1 above)
Comment (leopardracer) [reply docs/starknet/architecture/understanding-kakarot.md]: ok, I will fix it .

File: docs/starknet/ecosystem/sdks/foundry.md (hunk 2 above)
Comment (leopardracer) [reply docs/starknet/ecosystem/sdks/foundry.md:69]: nothing has changed here, the only thing is a typo

**Comment by leopardracer** [comment]
@Eikix Hello, I have made the requested fixes.

**Comment by leopardracer** [comment]
@Eikix Hi! Just wanted to remind you about this PR. If you have some time, please take a look. Thank you!

**Review by Eikix — APPROVED** [review]

**Pull request merged**